actors.md

```diff
@@ -688,26 +672,25 @@ func SubmitPost(proofs PoStProof, faults []FaultSet, recovered Bitfield, done Bi
 		TransferFunds(msg.From, msg.Value-feesRequired)
 	}
 
-	if !CheckPostProof(miner.SectorSize, proof, faults) {
+    seed := GetRandFromBlock(self.ProvingPeriodStart + ProvingPeriodDuration(self.SectorSize) - POST_CHALLENGE_TIME)
+    faultSet := self.faultSet
+
+	if !VerifyPoSt(self.SectorSize, self.provingSet, seed, proof, faultSet) {
```

Comment:
Something somewhere should document that this will fail if a fault has been declared at any point since `self.ProvingPeriodStart + ProvingPeriodDuration(self.SectorSize) - POST_CHALLENGE_TIME`

In case a miner does experience a fault after the challenge window starts, what mechanism will move their proving period forward to the subsequent window so that they can try again with a new fault set?
This is currently inconsistent with code higher up that prevents any PoSt submission after the generation attack time.

Reply:
This was actually tricker than I thought, I added more words, as well as more tracking to make this correct. please review again